packages/rich-text/src/to-dom.js

```diff
@@ -64,20 +67,38 @@ function append( element, child ) {
 	if ( typeof child === 'string' ) {
 		child = element.ownerDocument.createTextNode( child );
 	}
-
-	const { type, attributes } = child;
-
-	if ( type ) {
-		child = element.ownerDocument.createElement( type );
-
-		for ( const key in attributes ) {
-			child.setAttribute( key, attributes[ key ] );
-		}
+	if ( 'type' in child ) {
+		child = contextuallyCreate( element, child );
```

Comment:
I would leave the original code there and add one special case for if `type === 'svg'`. I thinlk the `contextuallyCreate` does not properly serve the situation: almost all the time we're just dealing with regular DOM elements.

For special types such as SVG or MathML we should create one function for each `createSVG`, `createMathML` etc.

Reply:
I tried this in 5c4be5b. I didn't make a `createSVG` function because I forgot about that part of your comment until now. Not sure if it's necessary at this stage but we can do that if you like.